File: Array/wuchao/array/7-28 寻找两个数组的交集/FindArrayIntersection.java
package com.slow.lib.practice.array;

import java.util.HashMap;
import java.util.Map;

/**
 * @Author wuchao
 * @Date 2020/7/28-10:06 AM
 * @description
 * @email 329187218@qq.com
 * @see
 */
public class FindArrayIntersection {
    public int[] intersection(int[] nums1, int[] nums2) {
        Map<Integer,Integer> maps = new HashMap();
        Map<Integer,Integer> maps2 = new HashMap();
        for(int item:nums1){
            maps.put(item,1);
        }
        for(int item:nums2){
            if(maps.containsKey(item)){
                maps2.put(item,1);
            }
        }
        int[] intersection = new int[maps2.size()];
        int i=0;
        for(int key:maps2.keySet()){
            intersection[i++] = key;
        }
        return intersection;
    }
}

File: Array/wuchao/array/7-28 寻找两个数组的交集/FindArrayIntersection.kt
package com.slow.lib.practice.array

import java.util.*

/**
 * @Author wuchao
 * @Date 2020/7/28-10:01 AM
 * @see
 * @description
 * @email 329187218@qq.com
 */
class FindArrayIntersection {
    fun intersection(nums1: IntArray, nums2: IntArray): IntArray {
        val maps1 = HashMap<Int,Int>()
        val maps2 = HashMap<Int,Int>()
        for(item in nums1){
            maps1[item] = 1
        }
        for(item in nums2){
            if(maps1.containsKey(item)){
                maps2[item] = 1
            }
        }
        val intersection = IntArray(maps2.size)
        var i = 0
        for(item in maps2.keys){
            intersection[i++] = item
        }
        return intersection
    }
}